contracts/allocators/LUSDAllocator.sol

```diff
@@ -224,55 +234,130 @@ contract LUSDAllocator is Ownable {
         frontEndAddress = _frontEndAddress; // address can be 0
 
         require(_wethAddress != address(0), "WETH token address cannot be 0x0");
-        weth = IERC20(_wethAddress);
+        weth = IWETH(_wethAddress);
+
+        require(_hopTokenAddress != address(0), "Hop Token address cannot be 0x0");
+        hopTokenAddress = _hopTokenAddress; // address can be 0
+
+        require(_uniswapV3Router != address(0), "UniswapV3Router address cannot be 0x0");
+        swapRouter = ISwapRouter(_uniswapV3Router);
+    }
+
+    /**
+        StabilityPool::withdrawFromSP() and LQTYStaking::stake() will send ETH here, so capture and emit the event
+     */
+    receive() external payable {
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    /* ======== CONFIGURE FUNCTIONS for Guardian only ======== */
+    function setPercentETHtoLUSD(uint8 _percentETHtoLUSD) external onlyGuardian {
+        require(_percentETHtoLUSD <= 100, "Percentage must be between 0 and 100");
+        percentETHtoLUSD = _percentETHtoLUSD;
+    }
+
+    function setPoolFee(uint24 _poolFee) external onlyGuardian {
+        require(_poolFee <= 100, "Pool fee must be between 0 and 100");
+        poolFee = _poolFee;
+    }
+
+    function setHopTokenAddress(address _hopTokenAddress) external onlyGuardian {
+        require(_hopTokenAddress != address(0), "Hop Token address cannot be 0x0");
+        hopTokenAddress = _hopTokenAddress;
+    }
+
+    /**
+     *  @notice setsFrontEndAddress for Stability pool rewards
+     *  @param _frontEndAddress address
+     */
+    function setFrontEndAddress(address _frontEndAddress) external onlyGuardian {
+        frontEndAddress = _frontEndAddress;
+    }
+
+    function updateTreasury() public onlyGuardian {
+        require(authority.vault() != address(0), "Zero address: Vault");
+        require(address(authority.vault()) != address(treasury), "No change");
```

Comment:
Why are we saying a reference to the treasury?

Reply:
Not sure was just blindly copying from https://github.com/OlympusDAO/olympus-contracts/blob/v2-allocators/contracts/allocators/ConvexAllocator.sol#L125-L129